annotationProcessor/src/main/java/org/corfudb/annotations/ObjectAnnotationProcessor.java

```diff
@@ -382,8 +382,28 @@ else if (
                     }
 
                     // If there is conflict information, calculate the conflict set
+                    boolean hasConflictData = false;
                     final String conflictField = "conflictField" + CORFUSMR_FIELD;
-                    if (m.hasConflictAnnotations) {
+
+
+                    // Check if there is a conflictFunction
+                    String conflictFunction = "";
+                    if (accessor != null && !accessor.conflictParameterFunction().equals("")) {
+                        conflictFunction = accessor.conflictParameterFunction();
+                    }
+                    if (mutator != null && !mutator.conflictParameterFunction().equals("")) {
+                        conflictFunction = mutator.conflictParameterFunction();
+                    }
+                    if (mutatorAccessor != null && !mutatorAccessor.conflictParameterFunction().equals("")) {
+                        conflictFunction = mutatorAccessor.conflictParameterFunction();
+                    }
+
+                    if (!conflictFunction.equals("")) {
+                        hasConflictData = true;
```

Comment:
Why don't you remove hasConflictAnnotations from SMRMethodInfo and replace it with hasConflictData and check if there exists either a conflict annotation or a conflict function. Wouldn't that be cleaner? 

Also, your if-else statement gives precedence to conflict functions over conflict annotations, if both exist. Is this by design ? maybe restrict that, so that you cant have both?